src/main/java/com/hbm/tileentity/machine/TileEntityFluidPump.java

```diff
@@ -0,0 +1,223 @@
+package com.hbm.tileentity.machine;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.hbm.interfaces.IFluidAcceptor;
+import com.hbm.interfaces.IFluidSource;
+import com.hbm.inventory.fluid.FluidType;
+import com.hbm.inventory.fluid.Fluids;
+import com.hbm.inventory.fluid.tank.FluidTank;
+import com.hbm.lib.Library;
+import com.hbm.tileentity.TileEntityMachineBase;
+
+import api.hbm.energy.IEnergyUser;
+import api.hbm.fluid.IFluidStandardTransceiver;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.common.util.ForgeDirection;
+import net.minecraft.block.Block;
+import net.minecraft.init.Blocks;
+import net.minecraft.world.World;
+
+public class TileEntityFluidPump extends TileEntityMachineBase implements IFluidAcceptor, IFluidSource, IEnergyUser, IFluidStandardTransceiver {
+	
+	public long power = 0;
+	public FluidTank[] tanks;
+	public List<IFluidAcceptor> list = new ArrayList();
+
+	public TileEntityFluidPump() {
+		super(0);
+		tanks = new FluidTank[1];
+		tanks[0] = new FluidTank(Fluids.WATER, 1000, 0);
+	}
+
+	@Override
+	public String getName() {
+		return "container.fluid_pump";
+	}
+
+	@Override
+	public void updateEntity() {
+		
+		if(!worldObj.isRemote) {
+			
+			this.updateConnections();
+			setupTanks();
+			
+			if(hasPower() && canPump() && tanks[0].getMaxFill() > tanks[0].getFill()) {
+				int amount = getPumpAmount();
+				int convert = Math.min(amount, tanks[0].getMaxFill() - tanks[0].getFill());
+				
+				if (convert > 0) {
+					tanks[0].setFill(tanks[0].getFill() + convert);
+					power -= this.getMaxPower() / 20;
```

Comment:
That's equivalent to 50 assemblers. Heavy infinite tanks are half as good and don't need any power at all.